pkg/kv/bulk/sst_batcher.go

```diff
@@ -193,6 +194,8 @@ type SSTBatcher struct {
 
 	asyncAddSSTs ctxgroup.Group
 
+	valueBuf bufalloc.ByteAllocator
```

Comment:
this is a this is just scratch space, right? this isn't really so much about amortizing lots of small per key allocations into a few rare big ones, but about reusing the _same buffer_ for every key, right?

Just making sure I'm reading it right.

I found bufalloc a little confusing for a pure scratch usecase, but maybe just naming the valScratch would have avoided that?